app/code/Magento/InventorySales/Observer/Product/IsProductSalable.php

```diff
@@ -0,0 +1,95 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\InventorySales\Observer\Product;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+use Magento\InventorySalesApi\Api\Data\SalesChannelInterface;
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+use Magento\InventorySalesApi\Api\StockResolverInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Class IsProductSalable
+ *
+ * @package Magento_InventorySales
+ */
+class IsProductSalable implements ObserverInterface
+{
+    /**
+     * @var IsProductSalableInterface
+     */
+    private $isProductSalable;
+
+    /**
+     * @var StockResolverInterface
+     */
+    private $stockResolver;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var array
+     */
+    private $stockIdByWebsite = [];
+
+    /**
+     * IsProductSalable constructor.
+     *
+     * @param IsProductSalableInterface $isProductSalable
+     * @param StockResolverInterface $stockResolver
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        IsProductSalableInterface $isProductSalable,
+        StockResolverInterface $stockResolver,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->isProductSalable = $isProductSalable;
+        $this->stockResolver = $stockResolver;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * Check if product is salable via MSI api
+     *
+     * @param Observer $observer
+     * @return void
+     */
+    public function execute(Observer $observer)
+    {
+        /** @var \Magento\Catalog\Model\Product $product */
+        $product = $observer->getProduct();
+        /** @var \Magento\Framework\DataObject $salableObject */
+        $salableObject = $observer->getSalable();
+
+        $websiteId = $product->getStore()->getWebsiteId();
+
+        $isSalable = $this->isProductSalable->execute($product->getSku(), $this->getStockId($websiteId));
+        $salableObject->setIsSalable($isSalable);
+    }
+
+    /**
+     * Retrieve stock_id by website
+     *
+     * @param int $websiteId The website Id
+     * @return int
+     */
+    private function getStockId($websiteId)
+    {
+        if (!isset($this->stockIdByWebsite[$websiteId])) {
+            $websiteCode = $this->storeManager->getWebsite($websiteId)->getCode();
+            $stock = $this->stockResolver->execute(SalesChannelInterface::TYPE_WEBSITE, $websiteCode);
```

Comment:
There are extensions out there which provide implementations for different Sales Channels then just Website. It's best to make the type and code parameters customizable from the outside, in this case it may be something like configuring object through the DI.